cli/tests/unit/fetch_test.ts

```diff
@@ -27,7 +27,20 @@ unitTest(
       async (): Promise<void> => {
         await fetch("http://localhost:4000");
       },
-      Deno.errors.Http,
+      TypeError,
+      "error trying to connect",
+    );
+  },
+);
+
+unitTest(
+  { perms: { net: true } },
+  async function fetchConnectionError(): Promise<void> {
+    await assertThrowsAsync(
+      async (): Promise<void> => {
+        await fetch("http://nil/");
```

Comment:
I think this URL is actually valid and doesn't cause a panic currently. Would be good to also test a really invalid URL like `http://<invalid>/`

Reply:
`http://nil/` cause panic from reqwest error, while `http://invalid/` cause panic because of Rust failing to parse invalid Uri. Thanks for adding another example. Will work on this too

Reply:
Apparently, deno_core::url doesn't throw error when parsing invalid Uri (neither did reqwest::Url). But error happen when reqwest trying to parse Url to http::Uri. 

This apparently can only be fixed by bumping up rust-url version to 2.2.